scripts/request_missing_tracks/request_missing_tracks.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Script for requesting missing tracks in a bundle"""
+
+import logging
+
+import typer
+from bundle import Bundle
+from juju import Juju, JujuFailedError
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+
+
+def get_charm_channel_map_for_applications(applications: dict[str, dict]) -> dict[str, dict]:
```

Comment:
You'd get the following error:

```
$ python3 ./request_missing_tracks.py bundle.yaml 
Traceback (most recent call last):
  File "./request_missing_tracks.py", line 17, in <module>
    def get_charm_channel_map_for_applications(applications: dict[str, dict]) -> dict[str, dict]:
TypeError: 'type' object is not subscriptable
```

I think you have to change this to something like

```suggestion
from typing import Dict
...
def get_charm_channel_map_for_applications(applications: Dict[str, dict]) -> Dict[str, dict]:
```

Tested on python3.8

Reply:
yep, I bet I was running in python 3.9! (I think 3.9 is where they made `dict` work that way)